common/views/components/ImageViewer/ImageViewer.js

```diff
@@ -46,48 +46,100 @@ const ImageViewer = ({
   infoUrl,
   src,
   srcSet,
+  manifest,
 }: ImageViewerProps) => {
+  const [pageIndex, setPageIndex] = useState(0);
   const [viewer, setViewer] = useState(null);
   const [isError, setIsError] = useState(false);
   const zoomStep = 0.5;
 
   useEffect(() => {
     if (viewer) {
-      fetch(infoUrl)
-        .then(response => response.json())
-        .then(data => {
-          viewer.open(getTileSources(data));
-        });
+      getLeftRightPageData().then(d => {
+        viewer.open(getTileSources(d));
+      });
     }
-  }, [infoUrl]);
+  }, [pageIndex]);
+
+  async function getLeftRightPageData() {
+    const leftInfoUrl = convertIiifUriToInfoUri(
+      manifest.sequences[0].canvases[pageIndex].images[0].resource.service[
+        '@id'
+      ]
+    );
+    const rightInfoUrl = convertIiifUriToInfoUri(
+      manifest.sequences[0].canvases[pageIndex + 1].images[0].resource.service[
+        '@id'
+      ]
+    );
+
+    const leftPromise = fetch(leftInfoUrl).then(res => res.json());
+    const rightPromise = fetch(rightInfoUrl).then(res => res.json());
+    const [left, right] = await Promise.all([leftPromise, rightPromise]);
+
+    return [left, right];
+  }
 
   async function setupViewer(imageInfoSrc, viewerId) {
+    const leftInfoUrl = convertIiifUriToInfoUri(
+      manifest.sequences[0].canvases[pageIndex].images[0].resource.service[
+        '@id'
+      ]
+    );
+    const rightInfoUrl = convertIiifUriToInfoUri(
+      manifest.sequences[0].canvases[pageIndex + 1].images[0].resource.service[
+        '@id'
+      ]
+    );
+
+    const leftPromise = fetch(leftInfoUrl).then(res => res.json());
+    const rightPromise = fetch(rightInfoUrl).then(res => res.json());
+    const leftAndRight = await Promise.all([leftPromise, rightPromise]);
+
     const { default: OpenSeadragon } = await import('openseadragon');
 
-    return fetch(imageInfoSrc)
-      .then(response => response.json())
-      .then(data => {
-        const osdViewer = OpenSeadragon({
-          id: `image-viewer-${viewerId}`,
-          showNavigationControl: false,
-          gestureSettingsMouse: {
-            scrollToZoom: false,
-          },
-          tileSources: getTileSources(data),
-        });
-        setIsError(false);
-        setViewer(osdViewer);
-
-        return osdViewer;
-      })
-      .catch(error => {
-        Raven.captureException(new Error(`OpenSeadragon error: ${error}`), {
-          tags: {
-            service: 'dlcs',
-          },
-        });
-        setIsError(true);
+    const osdViewer = OpenSeadragon({
+      id: `image-viewer-${viewerId}`,
+      showNavigationControl: false,
+      gestureSettingsMouse: {
+        scrollToZoom: false,
+      },
+      tileSources: getTileSources(leftAndRight),
+      collectionMode: true,
+      collectionRows: 1,
+    });
+    setIsError(false);
+    setViewer(osdViewer);
+
+    // Arrange two-up
+    OpenSeadragon.World.prototype.arrange = function() {
```

Comment:
Overwriting [the arrange method on the prototype](https://github.com/openseadragon/openseadragon/blob/81a634a61845aac973cf6b7d00ee0dcbbaf968ea/src/world.js#L321).